sklearn/base.py

```diff
@@ -335,6 +337,71 @@ def _get_tags(self):
                 collected_tags.update(more_tags)
         return collected_tags
 
+    def _check_n_features(self, X, reset):
+        """Set the n_features_in_ attribute, or check against it.
+
+        Parameters
+        ----------
+
+        X : ndarray or sparse matrix
+            The input samples
+        reset : bool
+            If True, the n_features_in_ attribute is set to X.shape[1]. Else,
+            the attribute must already exist and the function checks that it is
+            equal to X.shape[1].
+        """
+        n_features = X.shape[1]
+
+        if reset:
+            self.n_features_in_ = n_features
+        else:
+            if not hasattr(self, 'n_features_in_'):
```

Comment:
I try to avoid `try/except` when I can but that's personal pref.
I guess `hasattr` exists for a reason so we might as well just use it ^^
It might also be faster since it's written in C